### PR TITLE
feat: create delivery trip on draft notes

### PR DESCRIPTION
**Pull request by blaggacao**
#37307 wasn't intended for closure / abandonment.

`no-docs`

### CODE REVIEW

**Comment by blaggacao** [comment]
@deepeshgarg007 I think you wanted to [review this some time ago](https://github.com/frappe/erpnext/pull/37307#issuecomment-1807733289)?

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py [hunk 1]
```diff
@@ -72,6 +79,22 @@ def validate_stop_addresses(self):
 			if not stop.customer_address:
 				stop.customer_address = get_address_display(frappe.get_doc("Address", stop.address).as_dict())
 
+	def validate_delivery_note_not_draft(self):
+		delivery_notes = list(
+			set(stop.delivery_note for stop in self.delivery_stops if stop.delivery_note)
+		)
+		draft_delivery_notes = frappe.get_all(
+			"Delivery Note",
+			{"status": "Draft", "name": ["in", delivery_notes]},
```
Comment (barredterra) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.py]:
```suggestion
			{"docstatus": 0, "name": ["in", delivery_notes]},
```

The more reliable/standard way

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py [hunk 2]
```diff
@@ -72,6 +79,22 @@ def validate_stop_addresses(self):
...
+			pluck="name",
+		)
+		if draft_delivery_notes:
+			frappe.throw(
+				_(
+					"Delivery Notes may not be in Draft to submit. The following Delivery Notes are in draft {0}"
```
Comment (barredterra) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.py]:
```suggestion
					"Delivery Notes should not be in draft state when submitting a Delivery Trip. The following Delivery Notes are still in draft state: {0}. Please submit them first."
```

**Comment by blaggacao** [comment]
unstale

**Comment by blaggacao** [comment]
@barredterra could you help me reopen this and keep the stalebot at bay? I still intent to get this merged. It is important for certain use cases.

**Comment by blaggacao** [comment]
i think this also isn't intended for abandonment, either.

**Comment by blaggacao** [comment]
@deepeshgarg007 This is now ready, the backend code is fully unit tested and the use case has been evaluated and battle-tested for an extended period of time in our production deployment.

**Comment by blaggacao** [comment]
@deepeshgarg007 I fixed the oversight in CI that had escaped my local testing. From my side, this is ready to review/merge.

**Comment by blaggacao** [comment]
This is not intended to be closed.

**Comment by blaggacao** [comment]
@s-aga-r @rohitwaghchaure could someone please help out and engage with this PR? Thanks!

**Review by s-aga-r — APPROVED** [review]

**Pull request merged**